src/test/libcephfs/test.cc

```diff
@@ -655,6 +671,24 @@ TEST(LibCephFS, Xattrs_ll) {
   ceph_shutdown(cmount);
 }
 
+/**
+ * Test if sending dummy op crashes MDS or not.
+ */
+TEST(LibCephFS, check_dummy_op) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  UserPerm *perms = ceph_mount_perms(cmount);
+
+  int res = ceph_check_dummy_op(cmount, perms);
+  ASSERT_NE(res, 0);
```

Comment:
`ASSERT_EQ(res, -EOPNOTSUPP);`

?

Reply:
`auto reply = make_message<MClientSession>(CEPH_SESSION_REJECT);` , this is the code of it https://github.com/ceph/ceph/blob/main/src/client/Client.cc#L2407-L2417, so shouldn't it be `-EPERM`?

If you talk about the commit related to `reclaim_session` as well, there too `CEPHFS_EOPNOTSUPP` is switched to `CEPHFS_EINVAL`.

Reply:
Only place returning `-EOPNOTSUPP` is in `dispatch_client_request`. Can this request land here?

Reply:
`dispatch_client_request` returns `-CEPHFS_EOPNOTSUPP`. `ASSERT_EQ(res, -EOPNOTSUPP);` is the correct errno. Thanks.